libdnf-cli/progressbar/widgets/common.cpp

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright (C) 2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#include "common.hpp"
+
+#include <fmt/format.h>
+
+
+namespace libdnf::cli::progressbar {
+
+
+static const char * units[] = {
+    "B",
+    "kB",
+    "MB",
+    "GB",
+    "TB",
+    "PB",
+    "EB",
+    "ZB",
+    "YB",
+};
+
+
+std::string format_size(int64_t num) {
+    float i = static_cast<float>(num);
+    int index = 0;
+    while (i > 999) {
+        i /= 1024;
```

Comment:
Please add TODO. /1024 is used with metric units now.
/1024 -> KiB, MiB, GiB, TiB, ....
/1000 -> kB, MB, GB, TB, ...

Reply:
I changed the units according to https://en.wikipedia.org/wiki/Byte (to KiB etc.)
It's 2 more characters on the line, but I suppose we can live with that.